pmt/test/qa_pmt.cpp

```diff
@@ -99,4 +101,63 @@ TEST(Pmt, VectorWrites)
 
         EXPECT_EQ(int_pmt_vec->value(), int_vec_val_modified);
     }
-}
\ No newline at end of file
+}
+
+TEST(Pmt, VectorWrapper) {
+    pmt_vector_wrapper<uint32_t> x(10);
+    pmt_vector_wrapper<uint32_t> y{1,2,3,4,6,7};
+    std::vector<uint32_t> data{1,2,3,4,6,7};
+    for (size_t i = 0; i < y.size(); i++) {
+        EXPECT_EQ(y[i], data[i]);
+    }
+    // Make sure that range based for loop works.
+    size_t i = 0;
+    for (auto& e : y) {
+        EXPECT_EQ(e, data[i++]);
+    }
+
+    // Make sure I can mutate the data
+    for (auto& e: y) {
+        e += 2;
+    }
+    i = 0;
+    for (auto& e: y) {
+        EXPECT_EQ(e, data[i++]+2);
+    }
+
+    // Create from an std::vector
+    pmt_vector_wrapper<uint32_t> x_vec(std::vector<uint32_t>{1,2,3,4,6,7});
+
+    // Check the other constructors
+    pmt_vector_wrapper<uint32_t> vec1(4);
+    EXPECT_EQ(vec1.size(), 4);
+    for (auto& e: vec1)
+        EXPECT_EQ(e, 0);
+
+    pmt_vector_wrapper<uint32_t> vec2(4, 2);
+    for (auto& e: vec2)
+        EXPECT_EQ(e, 2);
+
+    pmt_vector_wrapper<uint32_t> vec3(data.begin(), data.end());
+    EXPECT_EQ(vec3.size(), data.size());
+    i = 0;
+    for (auto& e: vec3)
+        EXPECT_EQ(e, data[i++]);
+
+    pmt_vector_wrapper<uint32_t> vec4(vec3);
+    EXPECT_EQ(vec3.ptr(), vec4.ptr());
+}
+
+TEST(Pmt, MapWrapper) {
+    pmt_map2<std::string> x;
+    x["abc"] = 4;
+    x["qwer"] = std::vector<int>{1,2,4};
+    for (auto& [key, value]: x) {
+        std::cout << key << std::endl;
```

Comment:
Would be nice here to have some `EXPECT_EQ` tests and check against the value that is pulled out of the maps

Reply:
I can add that in.